src/Collection.php

```diff
@@ -418,7 +418,7 @@ public function createSearchIndex($definition, array $options = []): string
     public function createSearchIndexes(array $indexes, array $options = []): array
     {
         $operation = new CreateSearchIndexes($this->databaseName, $this->collectionName, $indexes, $options);
-        $server = select_server($this->manager, $options);
+        $server = select_server_for_write($this->manager, $options);
```

Comment:
I don't know it that's relevant for search indexes. 
> These commands internally proxy the search index management commands to a separate process that runs alongside an Atlas cluster. As such, read concern and write concern are not relevant for the search index management commands.
https://github.com/mongodb/specifications/blob/db3114e957f7c0976a1af09882dbb46cb4a70049/source/index-management/index-management.rst#where-are-read-concern-and-write-concern

Reply:
I'm familiar with `mongot`, but does that mean `mongod` has absolutely not state when it comes to search indexes? Perhaps that's the case, as `listSearchIndexes()` is distinct from `listIndexes()`.

This also raises the question of what happens if the search index operations are executed within a transaction. Presumably they don't maintain state on `mongod` at all (and thus operate outside of a transaction), but it's not clear to me if `mongod` would actually restrict their usage during an active transaction.

That said, I don't see the harm in requiring a primary in this case since it is a write operation from the user's perspective.